### ISSUE TITLE
Volumes in `ReactionSystem`

### ISSUE THREAD

**Issue by paulflang** [SciML/Catalyst.jl#347]
Is there a way we can store volumes in `ReactionSystem` (i.e. every species is associated with a volume/compartment it resides in)? This would allow us to optionally divide any simulation result by the volume, in case users are more interested in concentrations than absolute quantities.

**Comment by paulflang** [comment]
Oh sorry, I thought #312 was closed and wanted to start a cleaner discussion here. But it isn't so this and #312 are kind of duplicates.

**Comment by isaacsas** [comment]
I thought the resolution of the other issue is that this is better handled in SBML through generating rate constants within the ReactionSystem that are scaled appropriately by the volumes to have the desired units? 

How do you want the volumes to be handled internally? (i.e. are you looking for a specific assumed choice of units in each model, which I'd prefer to avoid, or for general units to be supported and volumes automatically handled in some way?)

**Comment by paulflang** [comment]
At the moment SBMLToolkit converts everything that is not already an absolute quantity to an absolute quantity. So in the resulting ReactionSystem contains species in absolute quantities and rate_rules of dimension 1/time.

There is not yet a notion of units SBMLToolkit atm (states are absolute quantities in arbitrary units, and time also is in arbitrary units), but @giordano is starting to implement them in SBML.jl. Perhaps he can comment on them.

**Comment by isaacsas** [comment]
So are you just looking for a place to store information like symbols for `(compartment,volume)`?

**Comment by giordano** [comment]
With https://github.com/LCSB-BioCore/SBML.jl/pull/109 we've got basic support for units: units defined in an SBML file are translated to [`Unitful.jl`](https://github.com/PainterQubits/Unitful.jl) units

**Comment by paulflang** [comment]
@isaacsas : Yes. Just to divide the solution for a species by the volume it resides in in case the user is more interested in concentrations than absolute numbers.

**Comment by ChrisRackauckas** [comment]
The concentration equations can just be included as observed variables, if we can come up with a good naming scheme for them. Then the user could just index them and get the solutions in the other variables, and that would all work in through a standard system.
